server/src/endpoints/minidump.rs

```diff
@@ -0,0 +1,161 @@
+            )
+        });
+
+    if name.is_none() && file_name.is_none() {
+        // log::trace!("multipart content without name or file_name");
+        return Box::new(ok(content));
+    } else {
+        let result = read_multipart_data(field, content.remaining_size).map(|data| {
+            content.remaining_size -= data.len();
+            match (name, file_name) {
+                (name, Some(file_name)) => {
+                    let mut item = Item::new(ItemType::Attachment);
+                    item.set_payload(ContentType::OctetStream, data);
+                    item.set_filename(file_name);
+                    name.map(|name| item.set_header("name", name));
+                    content.envelope.add_item(item)
+                }
+                (Some(name), None) => {
+                    let value = from_utf8(&data);
+                    match value {
+                        Ok(_value) => {
+                            let mut item = Item::new(ItemType::FormData);
+                            item.set_header("name", name);
+                            item.set_payload(ContentType::Text, data);
+                            content.envelope.add_item(item);
+                        }
+                        Err(_failure) => {
+
+                            // log::trace!("invalid text value in multipart item");
+                        }
+                    }
+                }
+                (None, None) => {
+                    unreachable!();
+                }
+            }
+            content
+        });
+        Box::new(result)
+    }
+}
+
+fn extract_envelope_from_multipart_request(
+    request: &HttpRequest<ServiceState>,
+    meta: EventMeta,
+    max_payload_size: usize,
+) -> ResponseFuture<Envelope, BadStoreRequest> {
+    let size_limited_envelope = SizeLimitedEnvelope {
+        envelope: Envelope::from_request(EventId::new(), meta),
+        remaining_size: max_payload_size,
+    };
+    Box::new(
+        handle_multipart_stream(size_limited_envelope, request.multipart())
+            .map(|size_limited_envelope| size_limited_envelope.envelope),
+    )
+}
+
+fn mini_dump_handler(
+    meta: EventMeta,
+    start_time: StartTime,
+    request: HttpRequest<ServiceState>,
+) -> ResponseFuture<HttpResponse, BadStoreRequest> {
+    Box::new(handle_store_like_request(
+        meta,
+        start_time,
+        request,
+        move |data, meta, max_event_payload_size| {
+            extract_envelope_from_multipart_request(data, meta, max_event_payload_size)
+        },
+        move |id| {
+            HttpResponse::Ok()
+                .content_type("text/plain")
+                .body(format!("{}", id))
+        },
+    ))
+}
+
+pub fn configure_app(app: ServiceApp) -> ServiceApp {
+    app.resource(r"/api/{project:\d+}/minidump/", |r| {
+        //hook security endpoint
+        r.method(Method::POST)
+            .filter(pred::Header("content-type", "multipart/form-data"))
```

Comment:
The minidump endpoint accepts raw payloads as well. Are we putting this out for later and are relying on the proxy endpoint for now? Might be worth a `// TODO`